monkey/monkey_island/cc/resources/agent_signals.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from http import HTTPStatus
+from json import JSONDecodeError
+
+from flask import request
+
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+from monkey_island.cc.services import AgentSignalsService
+
+logger = logging.getLogger(__name__)
+
+
+class AgentSignals(AbstractResource):
+    urls = ["/api/agent-signals/terminate-all", "/api/agent-signals/<string:agent_id>"]
+
+    def __init__(
+        self,
+        island_event_queue: IIslandEventQueue,
+        agent_signals_service: AgentSignalsService,
+    ):
+        self._island_event_queue = island_event_queue
+        self._agent_signals_service = agent_signals_service
+
+    def post(self):
+        try:
+            terminate_timestamp = request.json["kill_time"]
```

Comment:
Let's change "kill" to "terminate". We'll use "kill" if/when we implement a force-kill feature.